plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/apsadmin/portal/SeoPageAction.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2018-Present Entando Inc. (http://www.entando.com) All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.entando.entando.plugins.jpseo.apsadmin.portal;
+
+import com.agiletec.aps.system.services.lang.Lang;
+import com.agiletec.plugins.jacms.apsadmin.portal.PageAction;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import org.apache.commons.lang3.StringUtils;
+import org.entando.entando.plugins.jpseo.aps.system.services.metatag.Metatag;
+import org.entando.entando.plugins.jpseo.aps.system.services.page.PageMetatag;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.entando.entando.plugins.jpseo.aps.system.services.metatag.IMetatagCatalog;
+
+/**
+ * @author E.Santoboni
+ */
+public class SeoPageAction extends PageAction {
```

Comment:
Missing test coverage for this whole class

Reply:
In test class TestPageAction (in test package)